roles/ceph-config/templates/ceph.conf.j2

```diff
@@ -173,41 +173,41 @@ keyring = /var/lib/ceph/radosgw/{{ cluster }}-rgw.{{ hostvars[host]['ansible_hos
     {% endif %}
 {% elif radosgw_address is defined and radosgw_address != 'address' -%}
     {% if ip_version == 'ipv4' %}
-rgw frontends = civetweb port={{ radosgw_address }}:{{ radosgw_civetweb_port }} {{ radosgw_civetweb_options }}
+rgw frontends = {{ radosgw_frontend }} {% if radosgw_frontend == 'civetweb' -%}port{% elif radosgw_frontend == 'beast' -%}endpoint{%- endif -%}={{ radosgw_address }}:{{ radosgw_frontend_port }} {{ radosgw_frontend_options }}
     {% elif ip_version == 'ipv6' %}
-rgw frontends = civetweb port=[{{ radosgw_address }}]:{{ radosgw_civetweb_port }} {{ radosgw_civetweb_options }}
+rgw frontends = {{ radosgw_frontend }} {% if radosgw_frontend == 'civetweb' -%}port{% elif radosgw_frontend == 'beast' -%}endpoint{%- endif -%}=[{{ radosgw_address }}]:{{ radosgw_frontend_port }} {{ radosgw_frontend_options }}
     {% endif %}
 {% elif hostvars[host]['radosgw_interface'] is defined -%}
     {% set interface='ansible_' + (hostvars[host]['radosgw_interface'] | replace('-', '_')) %}
     {% if ip_version == 'ipv4' %}
-rgw frontends = civetweb port={{ hostvars[host][interface][ip_version]['address'] }}:{{ radosgw_civetweb_port }} {{ radosgw_civetweb_options }}
+rgw frontends = {{ radosgw_frontend }} {% if radosgw_frontend == 'civetweb' -%}port{% elif radosgw_frontend == 'beast' -%}endpoint{%- endif -%}={{ hostvars[host][interface][ip_version]['address'] }}:{{ radosgw_frontend_port }} {{ radosgw_frontend_options }}
     {% elif ip_version == 'ipv6' %}
-rgw frontends = civetweb port=[{{ hostvars[host][interface][ip_version][0]['address'] }}]:{{ radosgw_civetweb_port }} {{ radosgw_civetweb_options }}
+rgw frontends = {{ radosgw_frontend }} {% if radosgw_frontend == 'civetweb' -%}port{% elif radosgw_frontend == 'beast' -%}endpoint{%- endif -%}=[{{ hostvars[host][interface][ip_version][0]['address'] }}]:{{ radosgw_frontend_port }} {{ radosgw_frontend_options }}
     {% endif %}
 {% else %}
     {% set interface='ansible_' + (radosgw_interface | replace('-', '_')) %}
     {% if ip_version == 'ipv4' %}
-rgw frontends = civetweb port={{ hostvars[host][interface][ip_version]['address'] }}:{{ radosgw_civetweb_port }} {{ radosgw_civetweb_options }}
+rgw frontends = {{ radosgw_frontend }} {% if radosgw_frontend == 'civetweb' -%}port{% elif radosgw_frontend == 'beast' -%}endpoint{%- endif -%}={{ hostvars[host][interface][ip_version]['address'] }}:{{ radosgw_frontend_port }} {{ radosgw_frontend_options }}
     {% elif ip_version == 'ipv6' %}
-rgw frontends = civetweb port=[{{ hostvars[host][interface][ip_version][0]['address'] }}]:{{ radosgw_civetweb_port }} {{ radosgw_civetweb_options }}
+rgw frontends = {{ radosgw_frontend }} {% if radosgw_frontend == 'civetweb' -%}port{% elif radosgw_frontend == 'beast' -%}endpoint{%- endif -%}={{ radosgw_frontend_port }} {{ radosgw_frontend_options }}
```

Comment:
something happened to this line, it doesn't `hostvars[...]['address']` part in it

and instead of having the %if civetweb %elif beast stuff in each line, maybe do that once at the top and store the result in a variable like `endpoint_var`?